src/main/java/cz/muni/fi/timeline/entity/Teacher.java

```diff
@@ -1,15 +1,72 @@
 package cz.muni.fi.timeline.entity;
 
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
+import lombok.Getter;
+import lombok.Setter;
 
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Entity class that represents teacher.
+ * Teacher can own multiple study groups.
+ *
+ * @author Karolína Veselá
+ */
 
 @Entity
 public class Teacher {
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Getter
```

Comment:
Tu anotaci Getter můžeš dát před celou classu; a nemusíš to pak opakovat u každého fieldu.

Reply:
Nejspíš prostě dej Getter a Setter před celou classu. Nemyslím si, že je vyloženě špatně setter u id. Spíš bych ho tam ale dal:

> You may find that you may be able to get away without putting a getter/setter on the JPA Entity from the persistence side of things. But if you start dealing with entities that are serialized from other sources, even from your view in some cases, you will need a way to set the ID of the entity to let JPA know that it is dealing with an existing entity, if you cant set the id, then the persistence layer will just treat it as a new Object.